File: CS/TreeListCellMerging/TreeListForm.cs
using System;
using System.Windows.Forms;

namespace TreeListCellMerging
{
    public partial class TreeListForm : Form
    {
        public TreeListForm()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            this.departmentsTableAdapter.Fill(this.departmentsDataSet.Departments);
            treeList1.ExpandAll();
        }
        private void treeList1_AllowMergeRowCells(object sender, AllowMergeRowCellsEventArgs e)
        {
            if(e.Node.HasChildren == true)
                e.Merge = DevExpress.Utils.DefaultBoolean.True;
            else
                e.Merge = DevExpress.Utils.DefaultBoolean.Default;
        }
    }
}

File: VB/TreeListCellMerging/TreeListForm.vb
Imports System
Imports System.Windows.Forms

Namespace TreeListCellMerging
    Partial Public Class TreeListForm
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Me.departmentsTableAdapter.Fill(Me.departmentsDataSet.Departments)
            treeList1.ExpandAll()
        End Sub
        Private Sub treeList1_AllowMergeRowCells(ByVal sender As Object, ByVal e As AllowMergeRowCellsEventArgs) Handles treeList1.AllowMergeRowCells
            If e.Node.HasChildren = True Then
                e.Merge = DevExpress.Utils.DefaultBoolean.True
            Else
                e.Merge = DevExpress.Utils.DefaultBoolean.Default
            End If
        End Sub
    End Class
End Namespace